### ISSUE TITLE
Rewrite `bundled/tool/ruff_server.py` to typescript

### ISSUE THREAD

**Issue by T-256** [astral-sh/ruff-vscode#479]
IF we could convert this file into typescript, then extension wouldn't need Python installation/activation on target system at all.
Currently Ruff extension needs python interpreter to start the server.

_Originally posted by @T-256 in https://github.com/astral-sh/ruff-vscode/pull/443#discussion_r1611696376_

**Comment by T-256** [comment]
IMO, then we can also drop `ms-python.python` dependency since we don't need Python interpreter anymore. 

https://github.com/astral-sh/ruff-vscode/blob/da06307c428fe6e8c8dc5e720af6f4e3d9ad2239/src/extension.ts#L161-L164

At there we could use `ms-python.python` as optional dep instead of mandatory dep.

**Comment by charliermarsh** [comment]
Note that we have to call into Python if we want broad coverage for _all_ system Pythons. Otherwise, there's no way to (completely) reliably get the path to the scripts directory.

**Comment by T-256** [comment]
> I'm not sure if removing the `ms-python.python` dependency is desired in the long term, or at least, not without implementing our own python selector.
> 
> I suspect that we'll soon need to call into Python from Ruff to discover the module search paths and possibly more. But we could make `ms-python.python` an optional dependency.

Exactly, in addition I think, at this point Ruff itself has well defined its _config selector_ (based on Extension settings which then fallbacks to pyproject.toml/ruff.toml either on workspace dir or its parent dir). IIUC, in this extension, Python discovery is mainly used to pass target version of python to ruff. we could rely on `project.required-python` or `tool.ruff.target-version` instead.



> Otherwise, there's no way to (completely) reliably get the path to the scripts directory.

IIUC, we use scripts directory to find if Ruff binary available locally into that. This is good. What I suggested is to drop `ms-python.python` as _hard dependency_. which means if user already installed `ms-python.python` we can still get scripts directory and when not installed, this extension will lookup on $PATH only.

**Comment by dhruvmanila** [comment]
I think that script will still be important in the case when a user is

1. Opening VS Code for a project with a virtual environment
2. VS Code detects the virtual environment of the project
3. Ruff is installed in the virtual environment as a dev dependency (pinned at a specific version)

And, this seems like a pretty common scenario for a VS Code user. Other strategy cannot detect this. I think we should instead switch to https://github.com/microsoft/vscode-python-environments once it becomes stable.

**Comment by dhruvmanila** [comment]
> In this case Python extension already installed which then provides ability to user to select venv.
> When Python extension not installed, user cannot select environment, so it won't reach `find_ruff_binary_path.py` part. So, it will finally fallbacks to bundled version and works as expected from user perspective (no venv selected - so no need to use ruff executable in that venv).

That's a good point. I'll need to look closely but it is a possibility. If you're interested, feel free to look into it once `ruff-lsp` is completely removed which is currently scheduled with the `0.11` Ruff release.

> Note that I don't think it's doable for now and VSCode has or hasn't any relay to make it possible for integrations, just an idea in my mind to make it dependency-free of other extension-dependencies and do support integrations instead.

https://github.com/microsoft/vscode-python-environments could be useful although I haven't looked into it but it is advertised for this exact purpose (I think).
